src/main/java/uk/gov/digital/ho/hocs/casework/domain/repository/CaseDataRepository.java

```diff
@@ -25,4 +26,7 @@ public interface CaseDataRepository extends CrudRepository<CaseData, Long> {
     @Query(value = "SELECT cd.* FROM case_data cd WHERE cd.uuid = ?1", nativeQuery = true)
     CaseData findAnyByUuid(UUID uuid);
 
+    @Query(value="SELECT cast(cd.uuid as text) FROM case_data cd WHERE cd.migrated_reference = ?1 LIMIT 1", nativeQuery = true)
```

Comment:
Is there a check on the database column to ensure that only one case can have the same migrated_reference? If not, should there be? It will remove the need for the LIMIT 1.

Reply:
It can't be enforced in the database as E.g. UAT needs to support multiple migrated references as the allow duplicates flag is on there. There are existing QA cases with duplicates so this is needed to ensure a 409 is thrown rather than a 500 when Hibernate throws an exception due to unexpected multiple results.